sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Window.java

```diff
@@ -347,6 +353,21 @@ private void applicableTo(PCollection<?> input) {
               + " mode be specified using .discardingFiredPanes() or .accumulatingFiredPanes()."
               + " See Javadoc for more details.");
     }
+
+  }
+
+  private boolean allowedLatenessShorterThanUpstream(PCollection<?> input){
+    if (getAllowedLateness() != null
+        && getAllowedLateness().isShorterThan(input.getWindowingStrategy().getAllowedLateness())) {
+      LOG.warn(
```

Comment:
No need to warn any more, since it cannot cause data loss. You don't need this method any more.